### ISSUE TITLE
`atmos validate` ignores `ATMOS_CLI_CONFIG_PATH` env var

### ISSUE THREAD

**Issue by zdmytriv** [cloudposse/atmos#379]
### Describe the Bug

`atmos validate` ignores `ATMOS_CLI_CONFIG_PATH` env var

### Expected Behavior

`atmos validate` should respect `ATMOS_CLI_CONFIG_PATH` env var

### Steps to Reproduce

Validate works
```
 atmos validate component my-component-ue2 -s core-ue2-root
 ⧉  my-infra
```
 
Changing `ATMOS_CLI_CONFIG_PATH` and validating still works fine.
```
 ATMOS_CLI_CONFIG_PATH=/non/existing/path/to/atmos.yaml atmos validate component my-component-ue2 -s core-ue2-root
 ⧉  my-infra
```

**Comment by thejrose1984** [comment]
### Reproduction & code review

**TL;DR: Reproduced on current `main` (commit `30238462`).** Setting `ATMOS_CLI_CONFIG_PATH` to a non-existent path is silently ignored — Atmos does not error and falls back to config auto-discovery, so the command runs as if the env var were unset.

#### Environment
- Atmos built from `main` @ `30238462`, Go 1.26.3, darwin/arm64
- Run from `examples/quick-start-advanced` (which contains its own `atmos.yaml`)

#### Reproduction
| # | Command | Result |
|---|---------|--------|
| 1 (control) | `atmos validate component vpc -s plat-ue2-prod` | `✓ Component validated successfully` (exit 0) |
| 2 (bug) | `ATMOS_CLI_CONFIG_PATH=/non/existing/path atmos validate component vpc -s plat-ue2-prod` | `✓ Component validated successfully` (exit 0) — **identical to #1** |
| 3 (isolate) | as #2 but from a dir with **no** `atmos.yaml` | fails with the "Quick Start" no-config message |
| 4 (positive control) | from an empty dir, `ATMOS_CLI_CONFIG_PATH=<valid example dir>` | config loads and validation runs |

Test #2 is identical to the control, confirming the invalid path had **no effect** — matching the original report.

#### Root cause
`readEnvAmosConfigPath()` in `pkg/config/load.go` catches `viper.ConfigFileNotFoundError` and returns `nil`, so a missing file at `ATMOS_CLI_CONFIG_PATH` is treated as non-fatal:

```go
func readEnvAmosConfigPath(v *viper.Viper) error {
    atmosPath := os.Getenv(AtmosCliConfigPathEnvVar)
    if atmosPath == "" {
        return nil
    }
    err := mergeConfig(v, atmosPath, CliConfigFileName, true)
    if err != nil {
        var configFileNotFoundError viper.ConfigFileNotFoundError
        if errors.As(err, &configFileNotFoundError) {
            log.Debug("config not found ENV var "+AtmosCliConfigPathEnvVar, "file", atmosPath)
            return nil // <-- silently ignored
        }
        return err
    }
    ...
}
```

Additionally, when the env var is set, only git-root and parent-directory discovery are skipped; **CWD**, home (`~/.atmos`), and system configs are still loaded. So a valid `atmos.yaml` in the working directory is picked up and `validate` succeeds regardless of the bogus env var.

#### Assessment
The reported behavior is real and still present. Note it may be partly **by design** — every step in the discovery chain treats "file not found" as non-fatal, and there's even a test (`cmd/describe_component_test.go`) asserting that a nonexistent `ATMOS_CLI_CONFIG_PATH` produces only a downstream error, not a config-load error.

If the intent is for an **explicitly-set** `ATMOS_CLI_CONFIG_PATH` (and `--config-path`) to be authoritative, the fix would be to error out when the explicitly-requested path does not exist, rather than swallowing `ConfigFileNotFoundError` in `readEnvAmosConfigPath()` / `readAtmosConfigCli()`.

**Comment by osterman** [comment]
Resolved by #1868, which reworked base-path/config-file resolution and gave `ATMOS_CLI_CONFIG_PATH` proper precedence in the config search order (previously silently ignored). Closing — please reopen if you can still reproduce this on a recent version.
